### PR TITLE
Adds mocking api

### PR DESCRIPTION
**Pull request by ofhouse**
Okay, this took a few days longer due to the good weather here in Germany but finally finished it. 🙌
Hope you are also doing well during this time :smile: 

I suggest the following API for the mocking:

```js
export default {
  title: "QuoteCard",
  decorators: [withAngularJs("myApp")],
  parameters: {
    ng: {
      module: "myApp", // optional when provided in the decorator
      mock: {
        // When the app depends on modules which cannot be provided in the story you can mock them
        modules: ["some.external.module"],
        // You can mock / override constants here
        constants: {
          API_URL: "https://example.com",
        },
        // You can mock / override services here (dependency injection also works)
        services: {
          AppService: MockedAppService,
        },
        // You can mock / override factories here (dependency injection also works)
        factories: {
          LoggingService: mockLoggingService,
        },
      }
    },
  },
};
```

So it comes with the ability to mock the following things:
- **modules** `["moduleName1", "moduleName2", ...]`
Produces for each module provided a "fake" module with that name (So angular.js thinks the modules are loaded, but are in fact not)
- **constants** `{ CONSTANT_NAME: "value", ... }`
Provides / Replaces a constant with the provided key
- **services** `{ AppService: class AppService { }, ... }`
Provides / Replaces a service with the provided key
- **factories** `{ AppService: function appServiceFactory { return {} }, ... }`
Provides / Replaces a factory with the provided key

Also added a new module (`mock-test.module.js`) and story (`mocks.stories.js`) to test the API.

---

Additionally I removed the global loading of the app through the `preview.js` file and instead load the modules directly in the stories.

If you have any thoughts about the API design let me know I am open to suggestions.

### CODE REVIEW

File: example/stories/examples/mocks.stories.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+function MockedServiceFromFactory() {
+  let counter = 0;
+
+  return {
+    counter,
+    increaseCounter: () => {
+      return ++counter;
+    },
```
Comment (titonobre) [inline example/stories/examples/mocks.stories.js:34]: What do you think of this?

```suggestion
    get counter(){
      return counter;
    },
    increaseCounter: () => {
      return counter++;
    },
```

This allows the `MockedServiceFromFactory` to be used just like `MockedService` 

```html
<button ng-click="$ctrl.mockedServiceFromFactory.increaseCounter()">Increase Count</button>
<div>{{$ctrl.mockedServiceFromFactory .counter}}</div>
```

File: example/src/mock-test.module.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    <h2>Mocked Service</h2>
+    <div>
+      <button ng-click="$ctrl.mockedService.increaseCounter()">Increase Count</button>
+      <div>{{$ctrl.mockedService.counter}}</div>
+    </div>
+
+    <h2>Mocked Service (from factory)</h2>
+    <div>
+      <button ng-click="$ctrl.increaseFactoryCounter()">Increase Count</button>
+      <div>{{$ctrl.factoryCounter}}</div>
```
Comment (titonobre) [inline example/src/mock-test.module.js:25]: If suggestion to `MockedServiceFromFactory` is accepted
```suggestion
      <button ng-click="$ctrl.mockedServiceFromFactory.increaseCounter()">Increase Count</button>
      <div>{{$ctrl.mockedServiceFromFactory.counter}}</div>
```

File: example/src/mock-test.module.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  controller: class {
+    static $inject = ["MockedService", "MockedServiceFromFactory", "MOCKED_CONSTANT"];
+
+    constructor(mockedService, mockedServiceFromFactory, MOCKED_CONSTANT) {
+      this.mockedService = mockedService;
+      this.mockedServiceFromFactory = mockedServiceFromFactory;
+      this.MOCKED_CONSTANT = MOCKED_CONSTANT;
+
+      this.factoryCounter = 0;
```
Comment (titonobre) [inline example/src/mock-test.module.js:40]: If suggestion to `MockedServiceFromFactory` is accepted
```suggestion
```

File: example/src/mock-test.module.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    increaseFactoryCounter() {
+      this.factoryCounter = this.mockedServiceFromFactory.increaseCounter();
+    }
```
Comment (titonobre) [inline example/src/mock-test.module.js:45]: If suggestion to `MockedServiceFromFactory` is accepted
```suggestion
```

**Comment by ofhouse** [comment]
Nice, was looking for exactly what you suggested when I was writing the factory mock example. :+1: 
Was not aware that getters could also be used on plain objects thought this was only available in classes.
Thanks a lot for taking the time for doing a review!

**Pull request merged**